### PR TITLE
Create Vehicle Lights.md

### PR DESCRIPTION
**Pull request by cvray4242**

### CODE REVIEW

File: Documentation/Signoff/Vehicle Lights.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# Vehicle Lights signoff
+## Function of subsystem 
+The function of this subsystem is to allow users to interact with the car Headlights while also learning the concepts of electrical components and code. This subsystem will take different inputs at the same time from the input breadboard and Coding subsystems. It will use a microcontroller to take 4 inputs from those subsystems and send them to a multiplexer. The multiplexer will have an enable and two select lines that control what input is sent on through to the LEDs. The user will use inputs such as switches within the input breadboard subsystem to control the enable and select lines. 
+
+## Constraints
+### Derived from Shall statements
+
+C1: The Kit shall operate at safe and compatible voltage levels for educational purposes.
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Vehicle Lights.md]: Based on what definition? and what are those precisely?

File: Documentation/Signoff/Vehicle Lights.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+- The kit needs to operate at safe voltages but also make sure that the voltages are compatible with the Arduino and other components used in this subsystem since ratings and compatability can be different. 
+
+
+     
+
+C2: The kit shall ensure proper safety features for any connections that are made making sure they are connected properly and there are limitations for current so there is no overcurrent for safety reasons.
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Vehicle Lights.md]: What are those?

File: Documentation/Signoff/Vehicle Lights.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- All Arduino and soldered components need to be safely and properly connected. The soldered components must be soldered correctly and firmly. Any wires used to connect the Arduino to the multiplexer and LEDs that are soldered also need to be properly connected for safety protection. Any LEDs or other current-limited components will need to have current-limiting resistors to prevent over-current. 
+
+
+C3 & C9. The kit shall contain clear instructions, labeled components, user-friendly interface, and comprehensive materials such as a kit manual.
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Vehicle Lights.md]: How do you decide if something is or is not clearly labeled? (I'm being a bit pedantic here. But what I'm trying to get at is, a constraint needs to be precise enough that you can clearly show, through analysis, that it is or is not met.)

**Review by JesseTNRoberts — COMMENTED** [review]
Constraints need to be in a table and they need to be precise.

**Review by JesseTNRoberts — COMMENTED** [review]
I'm confused. Why is this discussing taking input from the various other systems if this is primarily about the lights?

**Review by JesseTNRoberts — COMMENTED** [review]
I don't think the function is to use a multiplexer. The function should not tell me how you intend to do something. Tell me what you are trying to do. 

In the schematic, I have no idea what any of the wires are doing. Things should be labeled.

**Review by JesseTNRoberts — COMMENTED** [review]
Why is this system only able to receive input from the resistance or capacitance? Why does this system care what is used to control it? Why not simply define a control input that is in percent?

I can't read the design. It is messy and spread out so that everything is far too small. Are these special LEDs that look like headlights? This design has to consider the physical look of the headlight as well.

File: Documentation/Signoff/Vehicle Lights.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+## Constraints
+Table. 1 Constraints
+|Description| Constraint | Source |
+|-|-|-|
+| Variable Type Range |The Arduino mega 2560 in this subsystem shall convert any ASCII variable type that ranges from 0 - 1,000,000 into a numeric variable type for better precision.| [2] C++ data types, https://www.w3schools.com/cpp/cpp_data_types.asp  |
+|Multiplexer Voltage Safety |The Multiplexer shall take 5 volts with a 10% tolerance, no more or less than that for this subsystem to prevent any damage to the multiplexer and ensure safe and sufficient operation.| Datasheet Requirement |
+|Current Safety |The LEDs in this subsystem shall take 15mA with a 5% tolerance, no more than that to prevent overcurrent damage and to ensure safe operation.| Datasheet Requirement  |
+| Socioeconomic | Any components and equipment within this subsystem should not exceed $55 dollars and the subsystems total cost should not exceed $175 to help keep the project budget in the 400-900 range| Conceptual Design|
```
Comment (JesseTNRoberts) [inline Documentation/Signoff/Vehicle Lights.md]: This isn't a socioeconomic constraint. This is a cost savings constraint. For it to be socioeconomic, it needs to be motivated by concerns for the economy or society.

**Review by JesseTNRoberts — COMMENTED** [review]
This much better! This is very close to being done.

I have a few questions that are probably better answered in person. 1) what is the multiplexer switching? What controls the multiplexer? 

Last, we need appropriate lights for a car. LEDs aren't really a good choice because they aren't directed.

**Review by JesseTNRoberts — COMMENTED** [review]
Are you sure that the current from the micro is sufficient to power the LEDs?

**Comment by micahr234** [comment]
You need to remove Jesse and request a review from me... when you are ready.

File: Documentation/Signoff/Vehicle Lights.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
```
Comment (micahr234) [inline Documentation/Signoff/Vehicle Lights.md]: Why are we using dip switches when we have a microprocessor and HMI? Would it not be easier to connect each resistor-LED network to a separate pin on the microprocessor?

File: Documentation/Signoff/Vehicle Lights.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
...ent and they do not consume much space. These LEDs will be sufficient to physically represent the headlights using white LEDs and tail lights using red LEDs to accurately represent the car lights. As you can see in Figure 4. each of the 4 LEDs will need its own small PCB with mounting holes so that they can be mounted at each corner of the car, to have a real representation of lights on a car. These LEDs will be small LEDs with a housing around them to help direct them into a beam like real car headlights. These LEDs are capable of operating on a voltage range of 3 volts to 7 volts allowing them to change brightness. These LEDs will be configured to an Arduino Atmega that is within the input system, the headlights will be connected to PWM pins D2, D3, D4, and D5 and will be coded so that they will operate as expected and also be able to interact with the HMI within the LCD block subsystem. Each LED-resistor network will have terminal blocks on the PCB as well so that the wire connected straight from the Arduino pins can be connected to the PCB and traced to the resistor/positive terminal of the LED and the ground wire can be connected and traced to the negative terminal of the LED.
```
Comment (micahr234) [inline Documentation/Signoff/Vehicle Lights.md]: Look up you leds. Do they fit in the circuit board footprint? Do they have a built in resistor?

File: Documentation/Signoff/Vehicle Lights.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
+# Vehicle Lights signoff
+## Function of subsystem 
+The function of this subsystem is to demonstrate Electrical and Computer Engineering concepts through physical means using the car's lighting system. This will allow users to interact with the car's lights and see the physical changes of the lights based on the user's input.  
+
+## Constraints
+Table. 1 Constraints
+|Description| Constraint | Source |
+|-|-|-|
+|Current Saftey / Standards |The LEDs in this subsystem shall take no more than 20 mA with 5% tolerance to prevent overcurrent damage and to ensure safe operation.| Datasheet Requirement  |
```
Comment (micahr234) [inline Documentation/Signoff/Vehicle Lights.md]: Did you address this in the analysis? What is the current draw of your LEDs?

**Review by micahr234 — APPROVED** [review]

**Pull request merged**